### PR TITLE
Fix #6 (Enable replaying from last processed event)

### LINKED ISSUE #6: Enable replaying from last processed event
web3-events will store last replayed event block number and array of log indexes. Then it will read from the file and register callbacks that will start from that block number, ignoring events with log indexes that were already processed. File name will be given in config.

### PR DESCRIPTION
**Pull request by jpmonettas**

### CODE REVIEW

File: src/district/server/web3_events.cljs [hunk 1]
```diff
@@ -6,16 +6,49 @@
...
             [mount.core :as mount :refer [defstate]]
-            [taoensso.timbre :as log]))
+            [taoensso.timbre :as log]
+            [cljs-node-io.core :as io :refer [slurp spit]]
+            [clojure.edn :as edn]))
 
 (declare start)
 (declare stop)
 
+(def checkpoint-a
```
Comment (madvas) [inline src/district/server/web3_events.cljs]: why is it called checkpoint-a with "a" at the end?

Comment (madvas) [inline src/district/server/web3_events.cljs]: I think cleaner would be if the atom is stored inside mount's web3-events atom. That's how we do it in other modules as well.

File: src/district/server/web3_events.cljs [hunk 2]
```diff
@@ -6,16 +6,49 @@
...
+  (swap! checkpoint-a
+         (fn [{:keys [last-processed-block processed-log-indexes] :as checkpoint-info}]
+           (-> checkpoint-info
+               (assoc :last-processed-block block-number
+                      :processed-log-indexes (if (= block-number last-processed-block)
+                                               (conj processed-log-indexes [tx log-index])
+                                               #{[tx log-index]}))))))
+
+
+(defn decorate-dispatch-callback [callback]
```
Comment (madvas) [inline src/district/server/web3_events.cljs]: I don't think this is correct naming. decorate is general term, yet the function decorates only with one specific function save-checkpoint. Therefore I think name of the function should be specific as well, such as `add-save-checkpoint`

**Review by madvas — CHANGES_REQUESTED** [review]

File: src/district/server/web3_events.cljs (hunk 1 above)
Comment (jpmonettas) [reply src/district/server/web3_events.cljs]: It is called checkpoint and is inside web3-events state now.

Comment (jpmonettas) [reply src/district/server/web3_events.cljs]: It is called checkpoint and is inside web3-events state now.

File: src/district/server/web3_events.cljs (hunk 2 above)
Comment (jpmonettas) [reply src/district/server/web3_events.cljs]: It is called wrap-callback-checkpoint-middleware now.

**Review by madvas — APPROVED** [review]

**Pull request merged**